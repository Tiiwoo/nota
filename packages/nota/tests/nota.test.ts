import * as cp from "child_process";
import os from "os";
import fs from "fs/promises";
import path from "path";
import util from "util";
import http from "http";

let exec = util.promisify(cp.exec);
let nota_path = path.join(process.cwd(), "packages", "nota", "dist", "nota.mjs");

let dir: string;
let input_path: string;
beforeAll(async () => {
  dir = await fs.mkdtemp(path.join(os.tmpdir(), "nota-"));
  input_path = path.join(dir, "index.nota");
  await fs.writeFile(input_path, `@h1{Hello world}`);

  let mod_path = path.join(process.cwd(), "packages", "nota");
  await exec(`yarn add ${mod_path}`, { cwd: dir });
});
afterAll(async () => {
  await fs.rm(dir, { recursive: true });
});

jest.setTimeout(30 * 1000);

test("nota-cli-builder", async () => {
  await exec(`node ${nota_path} build index.nota`, { cwd: dir });
  let html_path = path.join(dir, "dist", "index.html");
  let contents = await fs.readFile(html_path, "utf-8");
  expect(contents).toMatchSnapshot();
});

// test("nota-cli-server", async () => {
//   let p = cp.spawn(`node ${nota_path} edit index.nota --port 8765`, { cwd: dir, shell: true });
//   let once = (k: string) =>
//     new Promise(resolve => {
//       p.once(k, resolve);
//     });

//   await once("spawn");

//   // TODO: figure out http
//   // await new Promise(resolve =>
//   //   http.request(
//   //     {
//   //       hostname: "localhost",
//   //       port: 8765,
//   //     },
//   //     resolve
//   //   )
//   // );

//   p.kill('SIGINT');

//   await once("exit");
// });
