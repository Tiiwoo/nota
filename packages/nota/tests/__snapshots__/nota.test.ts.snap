// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nota-cli-builder 1`] = `"<!DOCTYPE html><html lang=\\"en\\" class=\\"\\"><head><link href=\\"index.css\\" rel=\\"stylesheet\\"><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"><title></title></head><body><div id=\\"root\\"><div id=\\"root\\"><div class=\\"nota-document\\"><div class=\\"nota-document-inner\\"><h1>Hello world</h1><div class=\\"logger\\"></div></div><div class=\\"portal\\"></div></div></div></div><script src=\\"./index.mjs\\" type=\\"module\\"></script></body></html>"`;
